### DISCUSSION TITLE
How can i get IL2CPP Method from method memory virtual address?

### DISCUSSION THREAD

**Discussion by NRTnarathip** [vfsfitvnm/frida-il2cpp-bridge#563] [Q&A] [answered]
Hi, currently i try to make stacktrace & print method info by method virtual address 
const methodAddress = DebugSymbol.fromAddress(address);
but i dont know how to get metadata about it
thanks.

**Comment by vfsfitvnm** [comment, marked as answer]
I do something similar here: https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/a28fa2eb5f10ef4a5c0635de1d9a11ec73a7231a/src/tracer.ts#L350-L363

What you could do is iterating over all the assemblies, images and classes and collect all the methods. Then, build an array and sort it by virtual address. Then, use binary search to _approximately_ get the method.
Keep in mind we only have the address of the first instruction of each method and we don't know how big a method is. So we assume a method instructions last until the next method begins. Of course this is imprecise.

However, there's definitely some metadata in the shared library. I recall seeing something interesting in `libil2cpp.so`, e.g. the method size of some methods. But I didn't investigate.

**Reply by NRTnarathip** [reply]
Cool it works fine, i hope in next update should have feature StackTrace like c# dotnet
🙏❤️

**Reply by vfsfitvnm** [reply]
Unfortunately I couldn't find a way to do that, and I'm not entirely sure whether it's even possible (in theory, it should). My knowledge is limited!
